A29-xds-tls-security.md

```diff
@@ -130,45 +153,124 @@ The CDS policy is the top level LB policy applied to all the connections under t
 [A27:CDS][] describes the flow. The field [`transport_socket`][CL-TS] is used to extract the
 [`UpstreamTlsContext`][UTC] as described [here][CL-TS-comment].
 Note that we don't (currently) support [`transport_socket_matches`][CL-TS-matches].
-The `UpstreamTlsContext` thus obtained is passed down to all the child policies and connections.
-How this is done is language dependent and is similar to how the implementations pass policy
-information down to child policies.
-
-[`common_tls_context`][CTC] in the `UpstreamTlsContext` contains the required security configuration.
-All other fields are ignored. See below for [`CommonTlsContext`][CTC-type] processing.
+The security configuration extracted from the `UpstreamTlsContext` thus obtained
+is passed down to all the child policies and connections. How this is done is language dependent
+and is similar to how the implementations pass policy information down to child policies.
+For example, C-core uses channel args to pass security configuration down, whereas
+Java constructs a dynamic `SslContextProvider` that is directly usable by a sub-channel's
+TLS handshaker to build an `SslContext` for the pending TLS handshake.
+
+Only when a channel is using `XdsChannelCredentials`, gRPC uses the security configuration
+in `UpstreamTlsContext`. [`common_tls_context`][CTC] in the `UpstreamTlsContext` contains
+the required security configuration. See below for [`CommonTlsContext`][CTC-type] processing.
+
+A secure client at a minimum requires [`validation_context_certificate_provider_instance`][VCCPI1]
+inside [`combined_validation_context`][CVC] to be able to validate the server certificate
+in the TLS mode. If this field is not set, gRPC will NACK the CDS update. The client identity
+certificate is configured via the [`tls_certificate_certificate_provider_instance`][TCCPI]
+field which is optional. The client certificate is sent to the server if the server requests
+or requires it in the TLS handshake. If the server requires the client certificate but is not
```

Comment:
Good question and I was actually thinking about this (whether I should include stuff that's not part of this spec) when I wrote this. 

The stuff about TLS handshake failing when client doesn't send the cert when server requires it, is not part of this spec but just informative about the mismatch that happens when client doesn't send the cert (since `tls_certificate_certificate_provider_instance` is not configured) even when server requires it. Let me see if I can reword to remove stuff that's not a requirement of this spec.